src/libraries/Microsoft.Extensions.Logging/src/LoggerFactoryScopeProvider.cs

```diff
@@ -220,7 +220,7 @@ IEnumerator IEnumerable.GetEnumerator()
             }
         }
 
-        private sealed class ActivityBaggageLogScopeWrapper : IEnumerable<KeyValuePair<string, string?>>
+        private sealed class ActivityBaggageLogScopeWrapper : IEnumerable<KeyValuePair<string, object?>>
```

Comment:
Isn't this a breaking change to remove an interface implementation that was previously available? Should we implement both IEnumerable interfaces?

Reply:
I don't think this is breaking. the worst case that if there is a provider which handle `IEnumerable<KeyValuePair<string, string?>>` and not handling `IEnumerable<KeyValuePair<string, object?>>`. This is very unlikely because they will not be handling most supported cases anyway and we are not seeing this in most providers we are aware of.

Reply:
> Do we make any specific guarantees in the BCL that exposed types won't change their interfaces? 

This is exposed as object but not as public API. 

> I think its more a question do we need to document this as a breaking change in behavior?

We can document it, but I am seeing this as a extremely low priority for now. We still have the option to document it as breaking charge if we see any complaint. I am fine if you think otherwise and should file breaking change anyway.

Reply:
If we don't have any previous rules around it I'm happy to wait and only do it in response to a complaint. Thanks!